### PR TITLE
Fix the debug modules model

### PR DESCRIPTION
**Pull request by echarles**
Fixes https://github.com/jupyterlab/jupyterlab/issues/11966

We need to respect the debug modules https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Modules

This PR updates the layout of the returned JSON (reply on the modules request) to respect the spec.

### CODE REVIEW

File: ipykernel/debugger.py [hunk 1]
```diff
@@ -600,11 +600,14 @@ async def modules(self, message):
...
-            filename = getattr(getattr(module, "__spec__", None), "origin", None)
-            if filename and filename.endswith(".py"):
-                mods.append({module.__name__: filename})
-
-        reply = {"body": {"modules": mods, "totalModules": len(modules)}}
+            filename = getattr(getattr(module, '__spec__', None), 'origin', None)
+            if filename and filename.endswith('.py'):
+                mods.append({
+                    'id': module.__name__,
+                    'name': filename
```
Comment (fcollonval) [inline ipykernel/debugger.py:607]: From the screenshot [there](https://github.com/jupyterlab/jupyterlab/issues/11966#issuecomment-1026031708), it seems more appropriate to use:

```python
{
  'id': <incremental integer>,
  'name': module.__name__,
  'path': filename
}
```

Comment (echarles) [reply ipykernel/debugger.py]: Done. Thx @fcollonval

**Comment by blink1073** [comment]
Kicking CI

**Review by blink1073 — APPROVED** [review]
Thanks!

**Comment by blink1073** [comment]
I'll hold off on a release for a bit in case we can get #849 included.

**Pull request closed without merging**